llvm/lib/Target/AMDGPU/SIInstructions.td

```diff
@@ -3003,6 +3060,40 @@ def : GCNPat <
   sub1)
 >;
 
+let True16Predicate = UseFakeTrue16Insts in
+def : GCNPat <
+  (i64 (bswap i64:$a)),
+  (REG_SEQUENCE VReg_64,
+  (V_BFI_B32_e64 (S_MOV_B32 (i32 0x00ff00ff)),
```

Comment:
Can you factor the original patterns into a class you can instantiate for each instruction type? This is a lot of tablegen to duplicate

Reply:
Hi Matt. The fake16 one is different from the original pattern since it has source modifiers so I think we cannot put them into one class. The t16 pattern can be merged with fake16 and I can do that in the t16 patch

Reply:
Can you conditionalize the use of the source modifier pattern in the input? something like `!if(SupportsSrcMods, (VOP3Mods $src0), $src0)`